webpack + require handlebars error

I get the following error simply by using require handlebars. 

``` text
▶ webpack --display-modules   
Hash: bdb0fe35e2f8bde783e5
Version: webpack 1.12.12
Time: 116ms
         Asset     Size  Chunks             Chunk Names
bundle.main.js  2.82 kB       0  [emitted]  main
   [0] ./src/index.js 150 bytes {0} [built]
   [1] ./~/handlebars/lib/index.js 792 bytes {0} [built] [3 warnings] [3 errors]

WARNING in ./~/handlebars/lib/index.js
require.extensions is not supported by webpack. Use a loader instead.

WARNING in ./~/handlebars/lib/index.js
require.extensions is not supported by webpack. Use a loader instead.

WARNING in ./~/handlebars/lib/index.js
require.extensions is not supported by webpack. Use a loader instead.

ERROR in ./~/handlebars/lib/index.js
Module not found: Error: Cannot resolve 'file' or 'directory' ../dist/cjs/handlebars/compiler/printer in /home/abhisekp/MyProjects/JSApps/MadeWithLove-Webpack/node_modules/handlebars/lib
 @ ./~/handlebars/lib/index.js 9:14-64

ERROR in ./~/handlebars/lib/index.js
Module not found: Error: Cannot resolve module 'fs' in /home/abhisekp/MyProjects/JSApps/MadeWithLove-Webpack/node_modules/handlebars/lib
 @ ./~/handlebars/lib/index.js 17:11-24

ERROR in ./~/handlebars/lib/index.js
Module not found: Error: Cannot resolve 'file' or 'directory' ../dist/cjs/handlebars in /home/abhisekp/MyProjects/JSApps/MadeWithLove-Webpack/node_modules/handlebars/lib
 @ ./~/handlebars/lib/index.js 7:17-50 
```

My **`index.js`**

``` js
var handlebars = require('handlebars');
```

My **`package.json`**

``` json
  "dependencies": {
    "handlebars": "^4.0.5"
  }
```

--- a/package.json
+++ b/package.json
@@ -74,10 +74,7 @@
   },
   "main": "lib/index.js",
   "types": "types/index.d.ts",
-  "browser": {
-    ".": "./dist/cjs/handlebars.js",
-    "./runtime": "./dist/cjs/handlebars.runtime.js"
-  },
+  "browser": "./dist/cjs/handlebars.js",
   "bin": {
     "handlebars": "bin/handlebars.js"
   },

--- a/tests/integration/webpack-test/package.json
+++ b/tests/integration/webpack-test/package.json
@@ -1,6 +1,6 @@
 {
   "name": "webpack-test",
-  "description": "Various tests with Handlebars and Webpack",
+  "description": "Various tests with Handlebars and multiple webpack versions",
   "version": "1.0.0",
   "main": "index.js",
   "scripts": {
@@ -8,14 +8,8 @@
     "test": "node dist/main.js"
   },
   "private": true,
-  "keywords": [],
-  "author": "",
-  "license": "ISC",
-  "dependencies": {},
-  "devDependencies": {
+  "dependencies": {
     "handlebars": "file:../../..",
-    "handlebars-loader": "^1.7.1",
-    "webpack": "^4.39.3",
-    "webpack-cli": "^3.3.7"
+    "handlebars-loader": "^1.7.1"
   }
 }

--- a/tests/integration/webpack-test/src/handlebars-esm-import-test.js
+++ b/tests/integration/webpack-test/src/handlebars-esm-import-test.js
@@ -1,0 +1,5 @@
+import Handlebars from 'handlebars/lib/handlebars';
+import { assertEquals } from './lib/assert';
+
+const template = Handlebars.compile('Author: {{author}}');
+assertEquals(template({ author: 'Yehuda' }), 'Author: Yehuda');

--- a/tests/integration/webpack-test/src/handlebars-runtime-test.js
+++ b/tests/integration/webpack-test/src/handlebars-runtime-test.js
@@ -1,0 +1,43 @@
+import * as Handlebars from 'handlebars/runtime';
+import { assertEquals } from './lib/assert';
+
+const template = Handlebars.template({
+  compiler: [8, '>= 4.3.0'],
+  main: function(container, depth0, helpers, partials, data) {
+    var helper,
+      lookupProperty =
+        container.lookupProperty ||
+        function(parent, propertyName) {
+          if (Object.prototype.hasOwnProperty.call(parent, propertyName)) {
+            return parent[propertyName];
+          }
+          return undefined;
+        };
+
+    return (
+      'Author: ' +
+      container.escapeExpression(
+        ((helper =
+          (helper =
+            lookupProperty(helpers, 'author') ||
+            (depth0 != null ? lookupProperty(depth0, 'author') : depth0)) !=
+          null
+            ? helper
+            : container.hooks.helperMissing),
+        typeof helper === 'function'
+          ? helper.call(depth0 != null ? depth0 : container.nullContext || {}, {
+              name: 'author',
+              hash: {},
+              data: data,
+              loc: {
+                start: { line: 1, column: 8 },
+                end: { line: 1, column: 18 }
+              }
+            })
+          : helper)
+      )
+    );
+  },
+  useData: true
+});
+assertEquals(template({ author: 'Yehuda' }), 'Author: Yehuda');

--- a/tests/integration/webpack-test/test.sh
+++ b/tests/integration/webpack-test/test.sh
@@ -2,15 +2,26 @@
 
 set -e
 
+run_tests () {
+  for i in dist/*-test.js ; do
+    echo "----------------------"
+    echo "-- Running $i"
+    echo "----------------------"
+    node "$i"
+    echo "Success"
+  done
+}
+
 # Cleanup: package-lock and "npm ci" is not working with local dependencies
 rm dist package-lock.json -rf
 npm install --legacy-peer-deps
-npm run build
 
-for i in dist/*-test.js ; do
-  echo "----------------------"
-  echo "-- Running $i"
-  echo "----------------------"
-  node "$i"
-  echo "Success"
-done
+# Test with webpack 4
+npm install --legacy-peer-deps --no-save webpack@^4 webpack-cli@^3
+npm run build
+run_tests
+
+# Test with webpack 5
+npm install --legacy-peer-deps --no-save webpack@^5 webpack-cli@^4
+npm run build
+run_tests

